Mathlib/NumberTheory/LSeries/HurwitzZetaEven.lean

```diff
@@ -395,3 +407,406 @@ lemma completedCosZeta₀_one_sub (a : UnitAddCircle) (s : ℂ) :
+  simp only [this, push_cast, one_mul] at h1
+  have h2 : Tendsto (fun s : ℂ ↦ s / 2) (𝓝[≠] 0) (𝓝[≠] (0 / 2)) :=
+    le_of_eq ((Homeomorph.mulRight₀ _ (inv_ne_zero (two_ne_zero' ℂ))).map_punctured_nhds_eq 0)
+  rw [zero_div] at h2
+  refine (h1.comp h2).congr (fun s ↦ ?_)
+  rw [completedHurwitzZetaEven, Function.comp_apply, div_mul_eq_mul_div, mul_div_assoc]
+
+lemma completedCosZeta_residue_zero (a : UnitAddCircle) :
+    Tendsto (fun s ↦ s * completedCosZeta a s) (𝓝[≠] 0) (𝓝 (-1)) := by
+  have h1 : Tendsto (fun s : ℂ ↦ s * _) (𝓝[≠] 0)
+    (𝓝 (-1)) := (hurwitzEvenFEPair a).symm.Λ_residue_zero
+  have h2 : Tendsto (fun s : ℂ ↦ s / 2) (𝓝[≠] 0) (𝓝[≠] (0 / 2)) :=
+    le_of_eq ((Homeomorph.mulRight₀ _ (inv_ne_zero (two_ne_zero' ℂ))).map_punctured_nhds_eq 0)
+  rw [zero_div] at h2
+  refine (h1.comp h2).congr (fun s ↦ ?_)
+  rw [completedCosZeta, Function.comp_apply, div_mul_eq_mul_div, mul_div_assoc]
+
+end FEPair
+
+/-!
+## Relation to the Dirichlet series for `1 < re s`
+-/
+
+/-- Formula for `completedCosZeta` as a Dirichlet series in the convergence range
+(first version, with sum over `ℤ`). -/
+lemma hasSum_int_completedCosZeta (a : ℝ) {s : ℂ} (hs : 1 < re s) :
+    HasSum (fun n : ℤ ↦ Gammaℝ s * cexp (2 * π * I * a * n) / (↑|n| : ℂ) ^ s / 2)
+    (completedCosZeta a s) := by
+  let c (n : ℤ) : ℂ := cexp (2 * π * I * a * n) / 2
+  have hF t (ht : 0 < t) : HasSum (fun n : ℤ ↦ if n = 0 then 0 else c n * rexp (-π * n ^ 2 * t))
+      ((cosKernel a t - 1) / 2) := by
+    refine ((hasSum_int_cosKernel₀ a ht).div_const 2).congr_fun fun n ↦ ?_
+    split_ifs with h <;>
+      simp only [zero_div, c, div_mul_eq_mul_div]
+  simp only [← Int.cast_eq_zero (α := ℝ)] at hF
+  rw [show completedCosZeta a s = mellin (fun t ↦ (cosKernel a t - 1 : ℂ) / 2) (s / 2) by
+    rw [mellin_div_const, completedCosZeta]
+    congr 1
+    refine ((hurwitzEvenFEPair a).symm.hasMellin (?_ : 1 / 2 < (s / 2).re)).2.symm
+    rwa [div_ofNat_re, div_lt_div_right two_pos]]
+  refine (hasSum_mellin_pi_mul_sq (zero_lt_one.trans hs) hF ?_).congr_fun fun n ↦ ?_
+  · apply (((summable_one_div_int_add_rpow 0 s.re).mpr hs).div_const 2).of_norm_bounded
+    intro i
+    simp only [c, (by { push_cast; ring } : 2 * π * I * a * i = ↑(2 * π * a * i) * I), norm_div,
+      RCLike.norm_ofNat, norm_norm, Complex.norm_exp_ofReal_mul_I, add_zero, norm_one,
+      norm_of_nonneg (by positivity : 0 ≤ |(i : ℝ)| ^ s.re), div_right_comm, le_rfl]
+  · simp only [c, Int.cast_eq_zero, ← Int.cast_abs, ofReal_intCast, div_right_comm, mul_div_assoc]
+
+/-- Formula for `completedCosZeta` as a Dirichlet series in the convergence range
+(second version, with sum over `ℕ`). -/
+lemma hasSum_nat_completedCosZeta (a : ℝ) {s : ℂ} (hs : 1 < re s) :
+    HasSum (fun n : ℕ ↦ if n = 0 then 0 else Gammaℝ s * Real.cos (2 * π * a * n) / (n : ℂ) ^ s)
+    (completedCosZeta a s) := by
+  have hs' : s ≠ 0 := (not_lt.mpr zero_le_one <| zero_re ▸ · ▸ hs)
```

Comment:
This is [Complex.ne_zero_of_one_lt_re](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Complex/Abs.html#Complex.ne_zero_of_one_lt_re).